src/lib/task.js

```diff
@@ -12,6 +12,7 @@ import fs from "mz/fs";
 import child_process from "mz/child_process";
 import taskcluster from 'taskcluster-client';
 import base from 'taskcluster-base';
+import promiseRetry from 'promise-retry';
```

Comment:
Wow, I didn't know about this library, but that certainly makes things easier!  @imbstack have you ever dealt with this library before? I'm not aware of the nuances (if any).

Reply:
It is used in artifact upload retries.

Reply:
Interesting! And nope, I've never used it before.